pkg/apis/flux/v1alpha1/validation/fluxconfig.go

```diff
@@ -122,3 +123,26 @@ func ValidateKustomization(kustomization *fluxv1alpha1.Kustomization, fldPath *f
 
 	return allErrs
 }
+
+// ValidateAdditionalSecretResources validates additionalResources
+func ValidateAdditionalSecretResources(additionalResources []fluxv1alpha1.AdditionalResource, shoot *gardencorev1beta1.Shoot, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if len(additionalResources) == 0 {
+		return allErrs
+	}
+	resources := map[string]gardencorev1beta1.NamedResourceReference{}
+	for _, resource := range shoot.Spec.Resources {
+		resources[resource.Name] = resource
+	}
```

Comment:
Can you dedupe this logic with `ValidateSource`?
Alternatively, we could also use `v1beta1helper.GetResourceByName`.

Reply:
deduped by using `GetResourceByName()`. This will iterate over the list every time, but i think it is fair to expect this list to never be too long to really matter.